assignment6.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+from statistics import median
+import unittest
+import numpy as np
+
+class RearrangingArray():
+
+	def __init__(self, arrangement, verbose=False):
+
+		self.arrangement = arrangement
+		self.swap_idx = None
+		self.order_dict = None
+		self.verbose = verbose
+
+	def _swap(self, i, j, k):
+		""" Swaps elements on indices i and j using an element on index k """
+		if self.verbose:
+			print(i, k)
+			print(i, j)
+			print(j, k)
+		self.arrangement[i],self.arrangement[k] = self.arrangement[k],self.arrangement[i]
+		self.arrangement[i],self.arrangement[j] = self.arrangement[j],self.arrangement[i]
+		self.arrangement[j],self.arrangement[k] = self.arrangement[k],self.arrangement[j]
+
+	def _move_swap_el(self, j):
+		""" Moves the swap element to position j from the current position """
+		i = self.swap_idx
+		while i < j:
+			if self.verbose:
+				print(i, i+1)
+			self.arrangement[i],self.arrangement[i+1]=self.arrangement[i+1],self.arrangement[i]
+			i += 1
+		while i > j:
+			if self.verbose:
+				print(i, i-1)
+			self.arrangement[i],self.arrangement[i-1]=self.arrangement[i-1],self.arrangement[i]
+			i -= 1
+		self.swap_idx = j
+
+	def _build_dict(self, new_order):
+		""" 
+		Builds a dictionary which defines the desired sorting order
+
+		Input:
+			new_order: List, a permutation of the class arrangement list
+		Output:
+			order_dict: Dict, keys are elements of the list, values are
+						positions in the sorted list
+		"""
+		order_dict = {}
+		for i, el in enumerate(new_order):
+			order_dict[el] = i
+		return order_dict
+
+	def _partition(self, l, r):
+		""" Partition function for QuickSort sorting algorithm """
+		pivot = median([
+			self.order_dict[self.arrangement[l]], 
+			self.order_dict[self.arrangement[(l+r)//2]], 
+			self.order_dict[self.arrangement[r]]]
+		)
+		while True:
```

Comment:
: D 
I also write while in that way, but as I know, this way of code style coders avoiding in job)